src/main/java/org/apache/sling/pipes/internal/ACLPipe.java

```diff
@@ -0,0 +1,260 @@
+    public static final String ACE_GRANT_KEY = "rep:GrantACE";
+    public static final String ACE_DENY_KEY = "rep:DenyACE";
+    public static final String JCR_PRIVILEGES_INPUT = "jcr:privileges";
+    public static final String PRIVILEGES_JSON_KEY = "privileges";
+
+    Session session;
+    ResourceResolver resolver;
+    UserManager userManager;
+    Privilege[] privileges;
+    String[] privilegesInput;
+    boolean allow;
+    boolean deny;
+    boolean authorizable;
+    Object outputBinding;
+
+    @Override
+    public Object getOutputBinding() {
+        if (outputBinding != null) {
+            return outputBinding;
+        }
+        return super.getOutputBinding();
+    }
+
+    @Override
+    public boolean modifiesContent() {
+        return allow || deny ;
+    }
+
+    /**
+     * public constructor
+     *
+     * @param plumber  plumber instance
+     * @param resource configuration resource
+     * @throws Exception bad configuration handling
+     */
+    public ACLPipe(Plumber plumber, Resource resource, PipeBindings upperBindings) throws Exception {
+        super(plumber, resource, upperBindings);
+        resolver = resource.getResourceResolver();
+        session = resolver.adaptTo(Session.class);
+        userManager = resolver.adaptTo(UserManager.class);
+        if (getConfiguration() != null) {
+            ValueMap properties = getConfiguration().adaptTo(ValueMap.class);
+            privilegesInput = properties.get(JCR_PRIVILEGES_INPUT, new String[]{});
+            allow = properties.get(PN_ALLOW, false);
+            deny = properties.get(PN_DENY, false);
+            authorizable = properties.get(PN_AUTHORIZABLE, false);
+        }
+    }
+
+    @Override
+    public Iterator<Resource> computeOutput() throws Exception {
+        Resource resource = getInput();
+        if (resource != null) {
+            if (allow || deny) {
+                logger.info("Going to changing ACL for the resource at path {}", resource.getPath());
+                if (StringUtils.isEmpty(getExpr())) {
+                    throw new Exception("expression for the principal or authorizable Id should be provided or provided correctly for privileges to be set");
+                }
+                Principal principal = getPrincipalFor(getExpr());
+                if ( null != privilegesInput && privilegesInput.length == 0 ){
+                    // create a privilege set with jcr:all
+                    privileges = allow ? AccessControlUtils.privilegesFromNames(session, Privilege.JCR_ALL) : AccessControlUtils.privilegesFromNames(session, Privilege.JCR_READ);
+                }
+                else {
+                    try {
+                        privilegesInput = processPrivilegesInput(privilegesInput);
+                        privileges = AccessControlUtils.privilegesFromNames(session, privilegesInput);
+                    }
+                    catch (Exception e) {
```

Comment:
i guess you don't want things to be caught, here right? if an exception happens it's better to get out of here. please either remove the catch, either re-throw